Changelog entry for new Smart Function block in public preview [PAB-5076] (#4156)
SoftwareAG/c8y-docs: content/change-logs/analytics/apama-in-c8y-20260129-Add-smart-function-block-in-public-preview.md

---
date: 2026-01-29

> title: Add Smart Function block in Public Preview

Reviewer: ```suggestion
title: Added Smart Function block in Public Preview
```

change_type:
  - value: change-pXAlHAWka
    label: Preview
product_area: Analytics
component:
  - value: component-M5-cepIIS
    label: Streaming Analytics
build_artifact:
  - value: tc-KXXmo2SUR
    label: apama-in-c8y
ticket: PAB-5076
version:
---

> A new `Smart Function` block has been added to Analytics Builder in Public Preview. This block executes a JavaScript Smart Function on the inputs and produces an output.

Reviewer: ```suggestion
A new Smart Function block has been added to Analytics Builder. This block executes a JavaScript Smart Function on the inputs and produces an output.

{{< c8y-admon-preview >}}
This feature is in Public Preview, that is, it is not enabled by default and may be subject to change in the future.
{{< /c8y-admon-preview >}}
```

Reviewer 2: Also, We shouldn't use the word "JavaScript", because it's trademarked by Oracle. Instead use "JS" or "ECMAScript"

---
date: 2026-01-29
title: Added Smart Function block in Public Preview
change_type:
  - value: change-pXAlHAWka
    label: Preview
product_area: Analytics
component:
  - value: component-M5-cepIIS
    label: Streaming Analytics
build_artifact:
  - value: tc-KXXmo2SUR
    label: apama-in-c8y
ticket: PAB-5076

> version:

Reviewer 3: Set version to `27.33.0` ... I see this is when https://github.com/Cumulocity-IoT/apama-in-c8y/pull/6423 was merged

Also remove the date as it will be auto populated

Author: Updated

---

A new Smart Function block has been added to Analytics Builder. This block executes a JS Smart Function on the inputs and produces an output.

{{< c8y-admon-preview >}}
This feature is in Public Preview, that is, it is not enabled by default and may be subject to change in the future.
{{< /c8y-admon-preview >}}

The Smart Function must export a function `onInput(inputs, context)` that returns a list of outputs.

**Argument details:**

* `inputs` is a list of ten `Value` objects, with members `value`, `properties`, and `timestamp` corresponding to the block inputs.

* `context` is an object with the following members:
	* `blockParameters` - List that contains the block parameters as specified in the block configuration.
	* `getState(key, def = null)` - Method that retrieves a value previously stored in the context under the given key. If no value is found, returns `def`.
	* `setState(key, value)` - Method that stores the given value in the context under the given key.

> In addition, the `console` object has `log`, `warn`, `error`, and `debug` members that can be used to log messages to the Analytics Builder log.

Reviewer 3: ```suggestion
In addition, the `console` object has `log`, `warn`, `error`, and `debug` members that can be used to log messages to the microservice log.
```
